polls/impl/src/java/org/sakaiproject/poll/service/impl/PollListManagerImpl.java

```diff
@@ -126,28 +127,16 @@ public void destroy() {
         List<Poll> polls = null;
         // get all allowed sites for this user
         List<String> allowedSites = externalLogic.getSitesForUser(userId, permissionConstant);
-        if (! allowedSites.isEmpty()) {
-            if (siteIds != null) {
-                if (siteIds.length > 0) {
-                    // filter down to just the requested ones
-                    for (int j = 0; j < allowedSites.size(); j++) {
-                        String siteId = allowedSites.get(j);
-                        boolean found = false;
-                        for (int i = 0; i < siteIds.length; i++) {
-                            if (siteId.equals(siteIds[i])) {
-                                found = true;
-                            }
-                        }
-                        if (!found) {
-                            allowedSites.remove(j);
-                        }
-                    }
-                } else {
-                    // no sites to search so EXIT here
-                    return new ArrayList<Poll>();
-                }
-            }
-            String[] siteIdsToSearch = allowedSites.toArray(new String[allowedSites.size()]);
```

Comment:
This is a strange and complex iteration, why not just:
`
List<String> requestedSiteIds = Arrays.asList(siteIds);
allowedSites.retainAll(requestedSiteIds);
`